server/container_create_linux.go

```diff
@@ -381,6 +395,50 @@ func (s *Server) addOCIBindMounts(ctx context.Context, ctr ctrfactory.Container,
 	return volumes, ociMounts, nil
 }
 
+// mountArtifact binds artifact blobs to the container filesystem based on the provided mount configuration.
+func (s *Server) mountArtifact(ctx context.Context, specgen *generate.Generator, m *types.Mount) ([]oci.ContainerVolume, error) {
+	artifact, err := s.ArtifactStore().Status(ctx, m.Image.Image)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get artifact status: %w", err)
+	}
+
+	paths, err := s.ArtifactStore().BlobMountPaths(ctx, artifact, s.config.SystemContext)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get artifact blob mount paths: %w", err)
+	}
```

Comment:
Do we want to combine them into one to fit more tightly into the use case?

Reply:
We want to use the artifact digest to look up when removing the artifact, so it'd be better to leave it as it is.

Reply:
Sounds good to me!